### ISSUE TITLE
Crash when running with `vitepress dev`

### ISSUE THREAD

**Issue by JasonWeinzierl** [ota-meshi/vite-plugin-eslint4b#126]
(I found this while trying to convert the vitepress site in `eslint-plugin-es-x` to TypeScript.)

When I run an app with `vitepress dev` with this plugin installed, ESLint fails in the browser, with this error:

```text
Uncaught (in promise) TypeError: process.cwd is not a function
    at normalizeCwd (__x00__virtual:eslint4b_eslint_linter:14404:51)
    at new Linter (__x00__virtual:eslint4b_eslint_linter:14440:10)
    at eslint-playground.vue:70:28
```

I also see this warning:

```text
Warning: Invalid input options (1 issue found)
- For the "inject". Invalid key: Expected never but received "inject". 
```

I realized that it's a problem with injecting the `process` shim into rolldown.  This is the current code:

```ts
  const build = await rolldown({
    input: entryPoints,
    external: externals,
    inject: {
      process: [
        path.normalize(path.join(dirname, "../shim/process-shim.mjs")),
        "*",
      ],
    },
  });
```

I think rolldown must have changed their API, because `inject` is not a property.  I think we must move `inject` into `transform`, like so:

```ts
  const build = await rolldown({
    input: entryPoints,
    external: externals,
    transform: {
      inject: {
        process: [
          path.normalize(path.join(dirname, "../shim/process-shim.mjs")),
          "*",
        ],
      },
    },
  });
```

I was able to temporarily patch my node_modules like this, and everything started working.

This doesn't seem to matter in actual production builds, so a published site is unaffected.  It's just local development.

**Comment by ota-meshi** [comment]
Thank you for reporting the issue and investigating the cause!

Could you please submit your changes as a pull request?
